src/parser.rs

```diff
@@ -342,6 +357,20 @@ impl<Impl: SelectorImpl> Debug for LocalName<Impl> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { self.to_css(f) }
 }
 
+impl<Impl: SelectorImpl> ToCss for SelectorList<Impl> {
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+        let mut iter = self.0.iter();
+        let first = iter.next()
+            .expect("Empty SelectorList, should contain at least one selector");
```

Comment:
I don’t think that would be appropriate, `fmt::Error` is for writing to the stream fails. And a selector list really should not be empty. (But that’s not worth enforcing with the type system.)